backend/src/api/message.py

```diff
@@ -0,0 +1,66 @@
+from database import get_session
+from dependencies import get_injector
+from fastapi import APIRouter, Depends, HTTPException, Request, status
+from schema.message_schema import MessageCreateRequest, MessageResponse
+from sqlalchemy.ext.asyncio import AsyncSession
+from usecase.channel_access_checker import ChannelAccessCheckerUseCaseIf
+from usecase.create_message import (
+    ChannelNotFoundError,
+    CreateMessageUseCaseError,
+    CreateMessageUseCaseIf,
+)
+from utils.api_utils import get_channel_access_checker
+from utils.logger_utils import get_logger
+
+# ロガーを取得
+logger = get_logger(__name__)
+
+router = APIRouter(prefix="/api")
+
+
+def get_usecase(injector=Depends(get_injector)) -> CreateMessageUseCaseIf:
+    return injector.get(CreateMessageUseCaseIf)
+
+
+async def check_channel_access_for_message(
+    req: MessageCreateRequest,
+    request: Request,
+    session: AsyncSession = Depends(get_session),
+    access_checker: ChannelAccessCheckerUseCaseIf = Depends(get_channel_access_checker),
+) -> None:
+    """メッセージ投稿時のチャンネルアクセス権限をチェックする依存関数"""
+    # 共通のアクセスチェッカーを使用してチャンネルアクセスを確認
+    await access_checker.execute(request, str(req.channel_id), session)
+
+
+# チャネルにメッセージを送信するエンドポイント
+@router.post("/message", response_model=MessageResponse, status_code=status.HTTP_201_CREATED)
```

Comment:
APIのエンドポイント設計的には複数形で書いてあるイメージが私の中では強いです。
このような外形的な指摘が今更になってしまってすみません。。。
```suggestion
@router.post("/messages", response_model=MessageResponse, status_code=status.HTTP_201_CREATED)
```